collector/mongod/top_counters.go

```diff
@@ -0,0 +1,79 @@
+package collector_mongod
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	topTimeSecondsTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: Namespace,
+		Name:      "top_time_seconds_total",
+		Help:      "The top command provides operation time, in seconds, for each database collection",
+	}, []string{"type", "database", "collection"})
+	topCountTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: Namespace,
+		Name:      "top_count_total",
+		Help:      "The top command provides operation count for each database collection",
+	}, []string{"type", "database", "collection"})
+)
+
+// TopStatsMap is a map of top stats
+type TopStatsMap map[string]TopStats
+
+// TopcountersStats topcounters stats
+type TopcounterStats struct {
+	Time  float64 `bson:"time"`
+	Count float64 `bson:"count"`
+}
+
+// TopCollectionStats top collection stats
```

Comment:
comment on exported type TopStats should be of the form "TopStats ..." (with optional leading article)